plugins/content-manager/src/main/java/com/wazuh/contentmanager/updater/ContentUpdater.java

```diff
@@ -0,0 +1,171 @@
+                this.restartConsumerInfo();
+                return;
+            }
+
+            currentOffset = nextOffset;
+        }
+
+        this.postUpdateCommand();
+    }
+
+    /**
+     * Fetches the context changes between a given offset range from the CTI API.
+     *
+     * @param fromOffset Starting offset (inclusive).
+     * @param toOffset Ending offset (exclusive).
+     * @return ContextChanges object containing the changes.
+     */
+    @VisibleForTesting
+    ContextChanges getContextChanges(String fromOffset, String toOffset) {
+        return Privileged.doPrivilegedRequest(
+                () -> CTIClient.getInstance().getChanges(fromOffset, toOffset, null));
+    }
+
+    /**
+     * Retrieves the latest offset from the CTI API.
+     *
+     * @return Latest available offset.
+     */
+    @VisibleForTesting
+    Long getLatestOffset() {
+        ConsumerInfo consumerInfo =
+                Privileged.doPrivilegedRequest(() -> CTIClient.getInstance().getCatalog());
+        return consumerInfo.getLastOffset();
+    }
+
+    /**
+     * Retrieves the current stored "last" offset from {@link ContextIndex}.
+     *
+     * @return The current "last" offset.
+     */
+    @VisibleForTesting
+    Long getCurrentOffset() {
+        ConsumerInfo consumerInfo =
+                Privileged.doPrivilegedRequest(
+                        () ->
+                                ContextIndex.getInstance()
+                                        .getConsumer(PluginSettings.CONTEXT_ID, PluginSettings.CONSUMER_ID));
+        return consumerInfo.getLastOffset();
+    }
+
+    /**
+     * Applies the fetched changes to the indexed context.
+     *
+     * @param changes Detected context changes.
+     * @return true if the changes were successfully applied, false otherwise.
+     */
+    @VisibleForTesting
+    boolean patchContextIndex(ContextChanges changes) {
+        // TODO: Call the patch method, return false on error.
+        return true;
+    }
+
+    /** Posts a new command to the Command Manager informing about the new changes. */
+    @VisibleForTesting
+    void postUpdateCommand() {
+        // Post new command informing the new changes.
+        Privileged.doPrivilegedRequest(
+                () -> {
+                    CommandManagerClient.getInstance()
+                            .postCommand(Command.create(getCurrentOffset().toString()));
+                    return null;
+                });
+    }
+
+    /** Resets the consumer info by setting its last offset to zero. */
+    @VisibleForTesting
+    void restartConsumerInfo() {
+        Privileged.doPrivilegedRequest(
+                () -> {
+                    ContextIndex.getInstance()
+                            .index(
+                                    new ConsumerInfo(
+                                            PluginSettings.CONSUMER_ID, PluginSettings.CONTEXT_ID, 0L, null));
+                    return null;
+                });
```

Comment:
Same here. I don't think it is needed to use privilege to execute a request to a non-protected index.

Reply:
For some reason it seems to be required, without the Privileged it raises this error. I'll do a further research
```
[2025-04-01T14:29:06,979][ERROR][c.w.c.i.ContextIndex     ] [integTest-0] Failed to retrieve context [vd_4.8.0], consumer [vd_1.0.0]: java.security.AccessControlException: access denied ("java.util.PropertyPermission" "opensearch.experimental.feature.tiered_remote_index.enabled" "read")
```